### PR TITLE
Proponent consent to feedback

### PR DESCRIPTION
**Pull request by sanjaytkbabu**
# Description

[PADS-233](https://apps.nrs.gov.bc.ca/int/jira/browse/PADS-233)
Proponent consent to feedback: 
1. added to submission intake form
2. added to submission form
3. migration created to add column to submission table

## Types of changes

New feature (non-breaking change which adds functionality) 

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/src/db/migrations/20240820000000_007-feedback-consent.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable max-len */
+import type { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+  return (
+    Promise.resolve()
+      // Alter public schema tables
+      .then(() =>
+        knex.schema.alterTable('submission', function (table) {
```
Comment (jujaga) [inline app/src/db/migrations/20240820000000_007-feedback-consent.ts:9]: Style: Lean towards using arrow functions over function declarations.

File: app/src/db/migrations/20240820000000_007-feedback-consent.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      )
+  );
+}
+
+export async function down(knex: Knex): Promise<void> {
+  return (
+    Promise.resolve()
+      // Drop consent_to_feedback column from submission table
+      .then(() =>
+        knex.schema.alterTable('submission', function (table) {
```
Comment (jujaga) [inline app/src/db/migrations/20240820000000_007-feedback-consent.ts:22]: Style: Lean towards using arrow functions over function declarations.

File: app/src/db/models/submission.ts [hunk 3]
```diff
@@ -26,6 +26,7 @@ export default {
       project_description: input.projectDescription,
       submitted_at: new Date(input.submittedAt ?? Date.now()),
       submitted_by: input.submittedBy,
+      consent_to_feedback: input.feedbackConsent,
```
Comment (kyle1morel) [inline app/src/db/models/submission.ts:29]: Let's try to keep variable naming as consistent as possible between the front end back ends when we can.
`feedbackConsent` -> `consentToFeedback`

File: frontend/src/components/housing/submission/SubmissionForm.vue [hunk 4]
```diff
@@ -399,6 +402,19 @@ onMounted(async () => {
...
+        label="Research opt-in"
+        :placeholder="props.submission.feedbackConsent ? 'Yes' : 'No'"
+        :disabled="!props.editable"
+        :options="['true', 'false']"
+        :option-label="
+          (e: string) => {
+            return e === 'true' ? BasicResponse.YES : BasicResponse.NO;
+          }
+        "
+      />
```
Comment (kyle1morel) [inline frontend/src/components/housing/submission/SubmissionForm.vue:417]: Could we not use the BasicResponse values as the options, and have line 203 check `submitData.feedbackConsent = values.feedbackConsent === BasicResponse.YES;` ?

**Review by wilwong89 — APPROVED** [review]

**Review by kyle1morel — APPROVED** [review]

**Pull request merged**